hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1203,7 +1212,21 @@ private static Stream<String> getColumnsToIndexWithoutRequiredMetas(HoodieMetada
   }
 
   private static Stream<String> getFirstNFieldNames(Schema tableSchema, int n) {
-    return getFirstNFieldNames(tableSchema.getFields().stream().map(Schema.Field::name), n);
+    return getFirstNFieldNames(tableSchema.getFields().stream()
+        .filter(field -> {
+          switch (resolveNullableSchema(field.schema()).getType()) {
```

Comment:
From what I can guage, these are the supported types 
```
  private static boolean canCompare(Schema schema, HoodieRecordType recordType) {
    // if recordType is SPARK then we cannot compare RECORD and ARRAY types in addition to MAP type
    if (recordType == HoodieRecordType.SPARK) {
      return schema.getType() != Schema.Type.RECORD && schema.getType() != Schema.Type.ARRAY && schema.getType() != Schema.Type.MAP;
    }
    return schema.getType() != Schema.Type.MAP;
  }
```

ref: https://github.com/apache/hudi/blob/5c28762c80c678e66df662c0f4e4c2855766840f/hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java#L1425 

Curious to understand from where did you get this list?

Reply:
Can we just re-use the same method. may be we can rename the method to, isDataTypeSupportedForColStats(Schema schema, HoodieRecordType recordType) 
and use it wherever required.